### PR TITLE
feat(vpn): add resource VPN client CA certificate

### PR DESCRIPTION
**Pull request by profoundwu**
**What this PR does / why we need it**:
add resource VPN client CA certificate
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add resource VPN client CA certificate
```

## PR Checklist

* [x] Tests added/passed.

```
make testacc TEST="./huaweicloud/services/acceptance/vpn" TESTARGS="-run TestAccClientCACertificate_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccClientCACertificate_basic -timeout 360m -parallel 4
=== RUN   TestAccClientCACertificate_basic
=== PAUSE TestAccClientCACertificate_basic
=== CONT  TestAccClientCACertificate_basic
--- PASS: TestAccClientCACertificate_basic (85.18s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       85.238s

```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
  image

   ab. Related resources (parent resources) not found
image

  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
 image

    bb. Related resources (parent resources) not found
image

### CODE REVIEW

**Comment by qigaoneng** [comment]
glmt

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_client_ca_certificate.go [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+			region := cfg.GetRegion(d)
+
+			var (
+				httpUrl = "v5/{project_id}/vpn-servers"
+				product = "vpn"
+			)
+			client, err := cfg.NewServiceClient(product, region)
+			if err != nil {
+				return nil, "ERROR", fmt.Errorf("error creating CFW client: %s", err)
+			}
```
Comment (houpeng80) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_client_ca_certificate.go]: it is unnecessary to create client for each time, it is better to be passed through param of the method

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_client_ca_certificate.go [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+
+			waitingRespBody, err := utils.FlattenResponse(waitingResp)
+			if err != nil {
+				return nil, "ERROR", err
+			}
+
+			findStatusStr := fmt.Sprintf("vpn_servers[?id == '%s']|[0].status", d.Get("vpn_server_id").(string))
+			statusRaw := utils.PathSearch(findStatusStr, waitingRespBody, nil)
+			if statusRaw == nil {
+				return nil, "ERROR", fmt.Errorf("error parsing %s from response body", `status`)
```
Comment (houpeng80) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_client_ca_certificate.go]:
```suggestion
				return nil, "ERROR", fmt.Errorf("error parsing status from response body")
```

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_client_ca_certificate.go [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+		},
+	}
+	_, err = updateClientCACertificateClient.Request("PUT", updateClientCACertificatePath, &updateClientCACertificateOpt)
+	if err != nil {
+		return diag.Errorf("error updating VPN client CA certificate: %s", err)
+	}
+
+	err = waitingForClientCACertificateStateCompleted(ctx, updateClientCACertificateClient, serverId, d.Timeout(schema.TimeoutUpdate))
+	if err != nil {
+		return diag.Errorf("error waiting for updating VPN client CA certificate (%s) to complete: %s", d.Id(), err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_client_ca_certificate.go]:
```suggestion
		return diag.Errorf("error waiting for updating VPN client CA certificate (%s) to complete: %s", id, err)
```

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_client_ca_certificate.go [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+	id := utils.PathSearch("client_ca_certificate.id", createClientCACertificateRespBody, "").(string)
+	if id == "" {
+		return diag.Errorf("unable to find client CA certificate ID in API response")
+	}
+	d.SetId(id)
+
+	serverId := d.Get("vpn_server_id").(string)
+	err = waitingForClientCACertificateStateCompleted(ctx, clientCACertificateClient, serverId, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.Errorf("error waiting for creating VPN client CA certificate (%s) to complete: %s", d.Id(), err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_client_ca_certificate.go]:
```suggestion
		return diag.Errorf("error waiting for creating VPN client CA certificate (%s) to complete: %s", id, err)
```

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_client_ca_certificate.go [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+		KeepResponseBody: true,
+	}
+	_, err = deleteClientCACertificateClient.Request("DELETE", deleteClientCACertificate, &deleteClientCACertificateOpt)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error deleting VPN Client CA certificate")
+	}
+
+	err = waitingForClientCACertificateStateCompleted(ctx, deleteClientCACertificateClient, serverId, d.Timeout(schema.TimeoutDelete))
+	if err != nil {
+		return diag.Errorf("error waiting for deleting VPN client CA certificate (%s) to complete: %s", d.Id(), err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_client_ca_certificate.go]:
```suggestion
		return diag.Errorf("error waiting for deleting VPN client CA certificate (%s) to complete: %s", id, err)
```

**Pull request merged**